test/SILGen/mangling.swift

```diff
@@ -184,3 +184,11 @@ func varargsVsArray(arr: [Int], n: String) { }
 
 // CHECK-LABEL: sil hidden [ossa] @$s8mangling14varargsVsArray3arr1nySaySiGd_SStF : $@convention(thin) (@guaranteed Array<Array<Int>>, @guaranteed String) -> ()
 func varargsVsArray(arr: [Int]..., n: String) { }
+
+// SWIFT_ENABLE_TENSORFLOW
+// CHECK-LABEL: sil hidden [ossa] @$s8mangling15funcVsDiffFunc12fnyS2fXE_tF : $@convention(thin) (@noescape @callee_guaranteed (Float) -> Float) -> ()
```

Comment:
Note: these newly added tests actually test `@differentiable` function mangling logic added in https://github.com/apple/swift/pull/25804.

That logic is different from the mangling logic added in this patch (mangling modifiers for `ImplFunctionType`). I'm not sure what's the meaning of `ImplFunctionType` and why it exists separately from `SILFunctionType` mangling: why is there `ImplEscaping` (for `ImplFunctionType`) in addition to `EscapingObjCBlock`/`EscapingAutoClosureType` (for regular `SILFunctionType`)?

cc @slavapestov, wonder if you can clarify